scripts/ziti-edge-tunnel-debug.bash

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env bash
+#
+# create a debug bundle for systemd service unit ziti-edge-tunnel.service
+#
+# usage: sudo /opt/openziti/bin/ziti-edge-tunnel-debug.bash [UPLOAD_URL]
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+checkCommand() {
+    if ! command -v "$1" &>/dev/null; then
+        echo "NOTICE: install '$1' and try again." >&2
+        $1
+    fi
+}
+
+main() {
+    # require commands
+    declare -a BINS=(sed tar timeout /opt/openziti/bin/ziti-edge-tunnel systemctl journalctl)
+    for BIN in "${BINS[@]}"; do
+        checkCommand "$BIN"
+    done
+
+    cd "$(mktemp -d)"
+    mkdir ./dump
+    chgrp -R ziti "${PWD}"
+    chmod -R g+rwX "${PWD}"
+    
+    echo -n "(estimated runtime 60s) [."
+    
+    NOW=$(date -u +'%Y-%m-%dT%H:%MZ')
+    ZITI_VERSION=$(/opt/openziti/bin/ziti-edge-tunnel version)
+    PREFIX=ziti-edge-tunnel-${ZITI_VERSION#v}-${NOW}
+    LOG_FILE=${PREFIX}.log
+    STRACE_FILE=${PREFIX}.strace
+    TUNNEL_STATUS_FILE=${PREFIX}.tunnel_status.json
+    SYSTEMD_RESOLVED_FILE=${PREFIX}.systemd-resolved
+    HOST_INFO_FILE=${PREFIX}.host
+    TARBALL=${PREFIX}.tgz
+    
+    # get the PID from systemd
+    ZET_PID="$(systemctl show -p MainPID --value ziti-edge-tunnel.service)"
+    if [[ -z "${ZET_PID}" ]] || [[ "${ZET_PID}" == "0" ]];
+    then
+        echo "ERROR: failed to get ziti-edge-tunnel.service PID" >&2
+        exit 1
+    fi
+    
+    # save the systemd-resolved state
+    (
+        set +e
+        set -x
+        systemctl status systemd-resolved.service
+        resolvectl status
+        resolvectl dns
+        resolvectl domain
+        ls -l /etc/resolv.conf
+        cat /etc/resolv.conf
+    ) &> "${SYSTEMD_RESOLVED_FILE}"
+    echo -n "."
+    
+    # save host info
+    (
+        set +e
+        set -x
+        hostnamectl; hostname
+        ip link show; ip addr show; ip route show
+        cat /etc/hosts /etc/*-release
+    ) &> "${HOST_INFO_FILE}"
+    echo -n "."
+    
+    # save the current service unit invocation's log messages
+    journalctl _SYSTEMD_INVOCATION_ID="$(systemctl show -p InvocationID --value ziti-edge-tunnel.service)" -l --no-pager \
+    &> "${LOG_FILE}"
+    echo -n "."
+
+    # if stack then save it; else try backtrace
+    # if [[ -f "/proc/${ZET_PID}/stack" ]]; then
+    if false; then
```

Comment:
is this testing leftover?

Reply:
Thanks for catching that. I failed to revert my temporary change for testing.